src/com/xilinx/rapidwright/util/RelocateDCPVertically.java

```diff
@@ -0,0 +1,392 @@
+/*
+ *
+ * Copyright (c) 2022 Xilinx, Inc.
+ * All rights reserved.
+ *
+ * Author: Pongstorn Maidee, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.xilinx.rapidwright.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.xilinx.rapidwright.design.ConstraintGroup;
+import com.xilinx.rapidwright.design.Design;
+import com.xilinx.rapidwright.design.Module;
+import com.xilinx.rapidwright.design.ModuleInst;
+import com.xilinx.rapidwright.design.Net;
+import com.xilinx.rapidwright.design.SiteInst;
+import com.xilinx.rapidwright.design.SitePinInst;
+import com.xilinx.rapidwright.device.PIP;
+import com.xilinx.rapidwright.device.Site;
+import com.xilinx.rapidwright.device.Tile;
+import com.xilinx.rapidwright.edif.EDIFHierNet;
+import com.xilinx.rapidwright.edif.EDIFNetlist;
+import com.xilinx.rapidwright.edif.EDIFPortInst;
+// Need util.Pair to allow compiling outside IDE with only dependent on RapidWright
+import com.xilinx.rapidwright.util.Pair;
+
+
+
+/**
+ * Fill some black boxes of a given design with a specific implementation.
+ */
+public class RelocateDCPVertically {
+
+
+	/**
+	 * Fill some black boxes of the given design with the given implementation.
+	 *
+	 * @param top         The design with black boxes to fill
+	 * @param mod         The implementation to fill the black boxes
+	 * @param cellAnchor The reference INT tile used as the handle to {@code mod} parameter
+	 * @param toCells     The list of pairs of a black box cell to be filled and its reference INT tile.
+	 *                    The x-coordinate of this INT tile must match that of the cellAnchor.
+	 */
+	public static boolean relocateCell(Design top, Module mod, String cellAnchor, List<Pair<String, String>> toCells) {
```

Comment:
```suggestion
	public static boolean placeModuleInsts(Design top, Module mod, String cellAnchor, List<Pair<String, String>> blackboxes) {
```

Reply:
"placeModuleInsts" may suggest something with placement (= look for a good location) and at the same time routing information is copied also. "insertModuleInsts" seems better fit to me. We could use fillBlackBoxes here, but I feel that information regarding black boxes can be left in the comment.

Reply:
I agree that `placeModuleInsts` maybe isn't the best description. Perhaps `relocateModuleInsts` then?